guacamole-ext/src/main/java/org/apache/guacamole/net/auth/simple/SimpleConnection.java

```diff
@@ -68,21 +72,34 @@ public SimpleConnection() {
      * @param identifier The identifier to associate with this connection.
      * @param config The configuration describing how to connect to this
      *               connection.
+     * 
+     * @throws GuacamoleException
+     *     If the default proxy configuration cannot be retrieved.
      */
     public SimpleConnection(String name, String identifier,
-            GuacamoleConfiguration config) {
+            GuacamoleConfiguration config) throws GuacamoleException {
```

Comment:
If possible, we should avoid adding new exceptions to an existing constructor, as downstream users of that constructor will suddenly have new things to catch.

Perhaps retrieving the configuration from the local environment could still be put off until `connect()`, being the default behavior if no `GuacamoleProxyConfiguration` is provided?

Reply:
Okay, will refactor.

Reply:
Backed this change out and just kept things in the `connect()` method.